src/conn/conn_log.c

```diff
@@ -660,22 +662,30 @@ __log_wrlsn_server(void *arg)
 	WT_CONNECTION_IMPL *conn;
 	WT_DECL_RET;
 	WT_SESSION_IMPL *session;
+	int yield;
 
 	session = arg;
 	conn = S2C(session);
+	yield = 0;
 	while (F_ISSET(conn, WT_CONN_LOG_SERVER_RUN)) {
 		/*
 		 * Write out any log record buffers.
 		 */
-		WT_ERR(__wt_log_wrlsn(session));
-		WT_ERR(__wt_cond_wait(session, conn->log_wrlsn_cond, 10000));
+		WT_ERR(__wt_log_wrlsn(session, &yield));
+		/*
+		 * If __wt_log_wrlsn did work we want to yield instead of sleep.
+		 */
+		if (yield++ < 1000)
```

Comment:
I don't understand why the count is here. Is there a reason to limit the number of times we yield to 1000?